fennel/testing/execute_aggregation.py

```diff
@@ -169,6 +170,43 @@ def get_val(self):
         return list(reversed(self.vals[-self.k :]))
 
 
+class FirstKState(AggState):
+    def __init__(self, k, dedup):
+        self.k = k
+        self.dedeup = dedup
+        self.vals = []
+
+    def add_val_to_state(self, val, _ts):
+        self.vals.append(val)
+        if not self.dedeup:
+            return list(self.vals[: self.k])
+        else:
+            to_ret = []
+            for v in self.vals:
+                if v not in to_ret:
+                    to_ret.append(v)
+                if len(to_ret) == self.k:
+                    break
+        return list(to_ret[: self.k])
+
+    def del_val_from_state(self, val, _ts):
+        if val in self.vals:
+            self.vals.remove(val)
+        if not self.dedeup:
+            return list(self.vals[: self.k])
+
+        ret = []
+        for v in self.vals:
+            if v not in ret:
+                ret.append(v)
+            if len(ret) == self.k:
+                break
+        return list(ret[: self.k])
+
+    def get_val(self):
+        return list(reversed(self.vals[-self.k :]))
```

Comment:
This has been copied as is from LastK. It should be `return list(self.vals[: self.k])`